src/main/api/2/studio-api-2.yaml

```diff
@@ -1325,6 +1327,105 @@ paths:
         '500':
           $ref: '#/components/responses/InternalServerError'
 
+  /webdav/list:
+    get:
+      tags:
+        - webdav
+      summary: Get a list of items from a WebDAV server
+      operationId: listItems
+      parameters:
+        - name: siteId
+          in: query
+          description: The site ID
+          required: true
+          schema:
+            type: integer
+            format: int64
+        - name: profileId
+          in: query
+          description: The profile ID
+          required: true
+          schema:
+            type: integer
+            format: int64
+        - name: path
+          in: query
+          description: The path of the directory to list (defaults to the root)
+          required: false
+          schema:
+            type: string
+        - name: type
+          in: query
+          description: The type of items to list (defaults to all types)
+          required: false
+          schema:
+            type: string
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  response:
+                    $ref: '#/components/schemas/ApiResponse'
+                  items:
+                    type: array
+                    items:
+                      $ref: '#/components/schemas/WebDavItem'
+        '401':
```

Comment:
Add 404

Reply:
Actually the WebDAV client never fails, it just returns an empty list